### PR TITLE
feat(tile): add content-top and content-bottom slots, deprecate content-start and content-end slots

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** #3462 

## Summary

This PR adds 2 new slots `content-top` and `content-bottom` to replace the now deprecated `content-start` and `content-end` slots.

### CODE REVIEW

File: packages/calcite-components/src/components/tile/tile.tsx [hunk 1]
```diff
@@ -8,11 +8,12 @@ import {
...
 import { CSS, SLOTS } from "./resources";
 import { Alignment, Scale } from "../interfaces";
-import { slotChangeHasAssignedElement } from "../../utils/dom";
 
 /**
- * @slot content-start - A slot for adding non-actionable elements before the component's content.
- * @slot content-end - A slot for adding non-actionable elements after the component's content.
+ * @slot content-top - A slot for adding non-actionable elements above the component's content.  Content slotted here will render in place of the `icon` property.
+ * @slot content-bottom - A slot for adding non-actionable elements below the component's content.
+ * @slot content-start - A slot for adding non-actionable elements before the component's content. @deprecated use `content-top` slot instead
```
Comment (jcfranco) [inline packages/calcite-components/src/components/tile/tile.tsx:15]: I _think_ `@deprecated` won't render as expected in the generated doc. cc @macandcheese Can you follow [this](https://github.com/Esri/calcite-design-system/blob/007dd64c9da4dc6bad73fb1f9822592bad1c4cd3/packages/calcite-components/src/components/card/card.tsx#L46) deprecation pattern for now?

File: packages/calcite-components/src/components/tile/tile.scss [hunk 2]
```diff
@@ -21,6 +21,7 @@
   min-inline-size: 140px;
 
   .container {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/tile/tile.scss:23]: @eriklharper Once merged, can you help consolidate styles in https://github.com/Esri/calcite-design-system/pull/8594/files#diff-e5512e4c4c1a21e32c7ace6b5d5a0dbf74427018fa79efbd26076e6a93cc3eda?

**Review by jcfranco — APPROVED** [review]
🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘
🤘🎸🎸🎸🎸🤘🎸🤘🤘🤘🤘🤘🎸🤘🎸🎸🎸🎸🤘🤘🎸🎸🎸🤘🤘🎸🎸🤘🤘🎸🤘🤘🤘🎸🤘🎸🎸🎸🎸🤘🎸🤘
🤘🎸🤘🤘🎸🤘🎸🤘🤘🤘🤘🤘🎸🤘🎸🤘🤘🤘🤘🎸🤘🤘🤘🤘🎸🤘🤘🎸🤘🎸🎸🤘🎸🎸🤘🎸🤘🤘🤘🤘🎸🤘
🤘🎸🎸🎸🎸🤘🎸🤘🤘🎸🤘🤘🎸🤘🎸🎸🎸🤘🤘🤘🎸🎸🤘🤘🎸🤘🤘🎸🤘🎸🤘🎸🤘🎸🤘🎸🎸🎸🤘🤘🎸🤘
🤘🎸🤘🤘🎸🤘🎸🤘🎸🤘🎸🤘🎸🤘🎸🤘🤘🤘🤘🤘🤘🤘🎸🤘🎸🤘🤘🎸🤘🎸🤘🤘🤘🎸🤘🎸🤘🤘🤘🤘🤘🤘
🤘🎸🤘🤘🎸🤘🤘🎸🤘🤘🤘🎸🤘🤘🎸🎸🎸🎸🤘🎸🎸🎸🤘🤘🤘🎸🎸🤘🤘🎸🤘🤘🤘🎸🤘🎸🎸🎸🎸🤘🎸🤘
🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘🤘

@ashetland @SkyeSeitz Could you help review the screenshot diffs? 🙇

File: packages/calcite-components/src/components/tile/tile.tsx (hunk 1 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/tile/tile.tsx:15]: Correct, the linked format for Card will render correctly - https://developers.arcgis.com/calcite-design-system/components/card/#slots

Comment (eriklharper) [reply packages/calcite-components/src/components/tile/tile.tsx:15]: Ah, thank you!  I was noticing weird stuff like this in VS Code too using the `@deprecated` syntax.  TS thinks I'm trying to deprecate the whole component, lol:

image

File: packages/calcite-components/src/components/tile/tile.scss (hunk 2 above)
Comment (eriklharper) [reply packages/calcite-components/src/components/tile/tile.scss:23]: PR ready for review here: https://github.com/Esri/calcite-design-system/pull/9012

**Comment by kevindoshier** [comment]
@jcfranco - we were using `content-end` to achieve this, with the action added to the end of the tile, is there a better way to do this now?

image

**Comment by geospatialem** [comment]
> @jcfranco - we were using `content-end` to achieve this, with the action added to the end of the tile, is there a better way to do this now?

@kevindoshier Would an `action` work in your use case? Perhaps something similar to this example: https://codepen.io/geospatialem/pen/JjVZgNL

image

**Comment by kevindoshier** [comment]
@geospatialem - kind of, what would be the best way to add the outline on the nested block?

image

**Comment by geospatialem** [comment]
> @geospatialem - kind of, what would be the best way to add the outline on the nested block?

@kevindoshier Updated the Codepen to include a nested block - would this work? https://codepen.io/geospatialem/pen/JjVZgNL 

image

**Comment by geospatialem** [comment]
> We should probably enhance Action to support `href`, `rel`, `target`, shouldn't have to use event listener for this kind of interaction.

Opened https://github.com/Esri/calcite-design-system/issues/9138

**Comment by kevindoshier** [comment]
@geospatialem - thanks!  That works!

**Pull request merged**